### PR TITLE
LRQA-77909 | master

### PR DESCRIPTION
**Pull request by Kvale1733**
https://issues.liferay.com/browse/LRQA-77909
https://issues.liferay.com/browse/LRQA-77910

Tested here:https://github.com/Kvale1733/liferay-portal/pull/216

### CODE REVIEW

**Comment by Kvale1733** [comment]
ci:test:sf

**Comment by Kvale1733** [comment]
ci:test:sf

File: build-test.xml [hunk 1]
```diff
@@ -4983,6 +4983,30 @@ tell application "Safari" to quit
 		</sequential>
 	</macrodef>
 
+	<macrodef name="prepare-test-environment-variables">
+		<attribute name="test.environment.variables" />
```
Comment (kyle-miho) [inline build-test.xml:4987]: @Kvale1733 you can just move the logic into the parent macro. There doesn't seem to a be a point to nest it into a macro here. Its probably more messy to create an extra target than to have a wrapper here

File: portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/ext/EnvironmentVariables.testcase [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+	@description = "Verify portal-ext properties can be set with environment variables after setting company-id-properties=true via jvm argument."
+	@priority = "3"
+	test ConfigurePortalExtPropertyViaJVM {
+		property test.environment.variables = "LIFERAY_PROPS_BY_COMPANY_20097=feature.flag.COMMERCE-8087=true${line.separator}CATALINA_OPTS=\"$CATALINA_OPTS -Dcompany-id-properties=true\"";
+		property test.name.skip.portal.instance = "EnvironmentVariables#ConfigurePortalExtPropertyViaJVM";
+
+		ImportExport.openImportExportAdmin();
+
+		ImportExport.gotoImport();
```
Comment (kyle-miho) [inline portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/ext/EnvironmentVariables.testcase:30]: We need a control check here in case this feature flag gets turned on, we will see it fail and know to change the feature flag, kind of like how you had a control check with the clamd anti-virus check.

It will also be more clean to move calling of the environment variables outside of the base server startup and just directly calling it from poshi since other tests will most likely never use this feature.

So basically -> Check FeatureFlag disabled -> Shutdown -> Enable Feature flag via Environment Variable ->Startup -> Check FeatureFlag enabled

**Comment by kyle-miho** [comment]
See comments
